### PR TITLE
Use latexmk instead of pdflatex for PDF generation

### PR DESCRIPTION
**Pull request by henrybear327**
The current Makefile is missing one more `pdflatex` pass.
    
The correct steps are : `pdflatex -> bibtex -> pdflatex -> pdflatex` as shown in the reference [here](https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-recipes). 

Otherwise, bib items will not be showing properly.

`.gitignore` has been updated accordingly.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,10 +2,8 @@ PROJ = lkmpg
 all: $(PROJ).pdf
 
 $(PROJ).pdf: lkmpg.tex
-	pdflatex -shell-escap $<
-	bibtex $(PROJ) >/dev/null || echo
-	pdflatex -shell-escape $< 2>/dev/null >/dev/null
 	rm -rf _minted-$(PROJ)
+	latexmk -shell-escape lkmpg.tex -pdf
```
Comment (jserv) [inline Makefile:6]: Add the guard before invoking `latexmk`. That is,
```
@if ! hash latexmk; then echo "No Latexmk found. See https://mg.readthedocs.io/latexmk.html for installation."; exit 1; fi 
```

**Review by jserv — CHANGES_REQUESTED** [review]
Update `README.md` for latexmk information

File: Makefile (hunk 1 above)
Comment (henrybear327) [reply Makefile:7]: Added as requested

**Comment by henrybear327** [comment]
> Update `README.md` for latexmk information

Updated as requested

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,9 @@ PROJ = lkmpg
 all: $(PROJ).pdf
 
 $(PROJ).pdf: lkmpg.tex
-	pdflatex -shell-escap $<
-	bibtex $(PROJ) >/dev/null || echo
-	pdflatex -shell-escape $< 2>/dev/null >/dev/null
 	rm -rf _minted-$(PROJ)
+	@if ! hash latexmk; then echo "No Latexmk found. See https://mg.readthedocs.io/latexmk.html for installation."; exit 1; fi
```
Comment (jserv) [inline Makefile:6]: Move the guard before `rm -rf`.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ $ brew install --cask mactex
 $ sudo tlmgr update --self
 ```
 
+The build system uses [latexmk](https://mg.readthedocs.io/latexmk.html) to generate PDF. It probably has been installed on your OS already. If not, please follow the [installation guide](https://mg.readthedocs.io/latexmk.html#installation).
```
Comment (jserv) [inline README.md:43]: `latexmk` is not a build system. Instead, it is a Perl script for running LaTeX the correct number of times to resolve cross references.
You can say, "latexmk is required to generated PDF."

File: Makefile (hunk 2 above)
Comment (henrybear327) [reply Makefile:6]: Changed as requested!

File: README.md (hunk 3 above)
Comment (henrybear327) [reply README.md:43]: Changed as requested!

**Pull request merged**